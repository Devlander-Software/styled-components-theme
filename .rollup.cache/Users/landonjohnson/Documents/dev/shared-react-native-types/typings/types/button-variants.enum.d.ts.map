{"version":3,"file":"button-variants.enum.d.ts","sourceRoot":"./","sources":["types/button-variants.enum.ts"],"names":[],"mappings":"AAAA,oBAAY,iBAAiB;IACzB,IAAI,SAAS;IACb,QAAQ,aAAa;IACrB,SAAS,cAAc;CAC1B"}