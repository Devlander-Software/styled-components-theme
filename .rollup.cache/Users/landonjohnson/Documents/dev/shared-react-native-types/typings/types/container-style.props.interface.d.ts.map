{"version":3,"file":"container-style.props.interface.d.ts","sourceRoot":"","sources":["../../src/types/container-style.props.interface.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,cAAc,EAAE,MAAM,eAAe,CAAA;AAC9C,OAAO,EAAE,yBAAyB,EAAE,6BAA6B,EAAE,MAAM,gCAAgC,CAAC;AAC1G,OAAO,EAAE,SAAS,EAAE,MAAM,cAAc,CAAC;AACzC,MAAM,WAAW,mBAAoB,SAAQ,SAAS,EAAE,6BAA6B;IACjF,WAAW,CAAC,EAAE,MAAM,CAAC;IACrB,YAAY,CAAC,EAAE,MAAM,CAAC;IACtB,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,YAAY,CAAC,EAAE,MAAM,CAAC;IACtB,OAAO,CAAC,EAAE,MAAM,CAAC;IACjB,IAAI,CAAC,EAAE,MAAM,CAAC;IAEd,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,UAAU,CAAC,EAAE,MAAM,CAAC;IACpB,UAAU,CAAC,EAAE,MAAM,CAAC;IACpB,WAAW,CAAC,EAAE,MAAM,CAAC;IACrB,aAAa,CAAC,EAAE,MAAM,CAAC;IACvB,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,QAAQ,CAAC,EAAE,MAAM,CAAC;IAClB,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,WAAW,CAAC,EAAE,MAAM,CAAC;CACxB;AAED,oBAAY,kBAAkB;IAC1B,MAAM,WAAW;IACjB,SAAS,eAAe;IACxB,OAAO,aAAa;IACpB,YAAY,kBAAkB;IAC9B,WAAW,iBAAiB;IAC5B,WAAW,iBAAiB;CAC/B;AAED,oBAAY,cAAc;IACtB,MAAM,WAAW;IACjB,SAAS,eAAe;IACxB,OAAO,aAAa;IACpB,OAAO,YAAY;IACnB,QAAQ,aAAa;CACxB;AAID,MAAM,WAAW,gCAAiC,SAAQ,mBAAmB,EAAE,yBAAyB;IACpG,KAAK,EAAE,cAAc,CAAC;CACzB"}