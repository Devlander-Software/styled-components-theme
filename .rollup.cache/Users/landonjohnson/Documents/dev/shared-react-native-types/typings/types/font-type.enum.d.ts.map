{"version":3,"file":"font-type.enum.d.ts","sourceRoot":"./","sources":["types/font-type.enum.ts"],"names":[],"mappings":"AAAA,oBAAY,YAAY;IACpB,KAAK,UAAU;IACf,KAAK,UAAU;IACf,KAAK,UAAU;IACf,aAAa,kBAAkB;CAClC"}