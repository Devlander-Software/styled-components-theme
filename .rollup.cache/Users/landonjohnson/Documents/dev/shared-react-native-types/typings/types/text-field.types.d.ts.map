{"version":3,"file":"text-field.types.d.ts","sourceRoot":"./","sources":["types/text-field.types.ts"],"names":[],"mappings":"AAAA,oBAAY,kBAAkB;IAC1B,IAAI,SAAS;IACb,KAAK,UAAU;IACf,SAAS,cAAc;IACvB,UAAU,eAAe;CAC1B;AAEH,oBAAY,gBAAgB;IACxB,OAAO,YAAY;IACnB,SAAS,eAAe;IACxB,UAAU,gBAAgB;IAC1B,OAAO,YAAY;IACnB,YAAY,kBAAkB;IAC9B,QAAQ,cAAc;CACvB;AAEH,oBAAY,gBAAgB;IACxB,IAAI,SAAS;IACb,GAAG,QAAQ;IACX,WAAW,gBAAgB;IAC3B,IAAI,SAAS;IACb,aAAa,kBAAkB;CAChC;AACH,oBAAY,mBAAmB;IAC3B,IAAI,SAAS;IACb,GAAG,QAAQ;IACX,WAAW,gBAAgB;IAC3B,mBAAmB,wBAAwB;IAC3C,YAAY,iBAAiB;IAC7B,WAAW,gBAAgB;IAC3B,gBAAgB,qBAAqB;IACrC,YAAY,iBAAiB;IAC7B,UAAU,eAAe;IACzB,iBAAiB,sBAAsB;IACvC,SAAS,cAAc;IACvB,QAAQ,aAAa;IACrB,QAAQ,aAAa;IACrB,UAAU,eAAe;IACzB,IAAI,SAAS;IACb,UAAU,eAAe;IACzB,UAAU,eAAe;IACzB,QAAQ,aAAa;IACrB,gBAAgB,qBAAqB;IACrC,UAAU,eAAe;IACzB,kBAAkB,uBAAuB;IACzC,kBAAkB,uBAAuB;IACzC,WAAW,gBAAgB;IAC3B,eAAe,oBAAoB;IACnC,QAAQ,aAAa;IACrB,WAAW,gBAAgB;IAC3B,WAAW,gBAAgB;CAC5B;AAEH,oBAAY,oBAAoB;IAC5B,eAAe,IAAA;IACf,gBAAgB,IAAA;IAChB,iBAAiB,IAAA;IACjB,gBAAgB,IAAA;IAChB,QAAQ,IAAA;IACR,QAAQ,IAAA;IACR,YAAY,IAAA;IACZ,cAAc,IAAA;IACd,aAAa,IAAA;IACb,WAAW,IAAA;IACX,OAAO,KAAA;IACP,QAAQ,KAAA;IACR,MAAM,KAAA;IACN,aAAa,KAAA;IACb,YAAY,KAAA;IACZ,aAAa,KAAA;IACb,qBAAqB,KAAA;IACrB,aAAa,KAAA;IACb,aAAa,KAAA;IACb,UAAU,KAAA;IACV,cAAc,KAAA;IACd,gBAAgB,KAAA;IAChB,wBAAwB,KAAA;IACxB,yBAAyB,KAAA;IACzB,qCAAqC,KAAA;IACrC,yBAAyB,KAAA;IACzB,uBAAuB,KAAA;IACvB,aAAa,KAAA;IACb,gBAAgB,KAAA;IAChB,SAAS,KAAA;IACT,KAAK,KAAA;IACL,kBAAkB,KAAA;IAClB,cAAc,KAAA;IACd,YAAY,KAAA;IACZ,UAAU,KAAA;IACV,cAAc,KAAA;IACd,KAAK,KAAA;IACL,SAAS,KAAA;CACV"}