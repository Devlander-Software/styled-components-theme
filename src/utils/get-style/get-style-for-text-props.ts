import { FontTypeEnum } from "../../types/font-type.enum"
import { TextStylePropsWithTheme } from "../../types/text-style.props"
import { capFontSize } from './cap-font-size';


export const getStyleForTextProps = (props: TextStylePropsWithTheme): string => {
	const { theme, fontSize,maxFontSize, textColorFromTheme, fontType, fontTypeWeight, fontWeight, ...restProps } = props

	const handleUnitProps = theme?.handleUnitProps ?? null
	const handleColorFromTheme = theme?.handleColorFromTheme ?? null
	const handleFontSizeProps = theme?.handleFontSizeProps ?? null
	const handleFontFromTheme = theme?.handleFontFromTheme ?? null
	const capFontSizeFunc = theme?.capFontSize ?? capFontSize

	const getColor = () => {
		if (restProps.onDark) {
			return theme.colors.whiteAlpha100
		}
		if (restProps.highlight) {
			return theme.colors.accent
		}
		if (textColorFromTheme && theme.colors[textColorFromTheme]) {
			return theme.colors[textColorFromTheme]
		}
		if (restProps.destructive) {
			return theme.colors.errorColor
		}
		return theme.colors.blackAlpha100
	}

	const cssProperties = [

		fontType && fontTypeWeight && handleFontFromTheme ? `font-family: ${handleFontFromTheme(fontType, fontTypeWeight, theme)};` : `font-family: ${handleFontFromTheme(FontTypeEnum.Font1, "bold", theme)};`,
		restProps.textDecorationLine && restProps.textDecorationLine !== "none" ? `text-decoration-line: ${restProps.textDecorationLine};` : null,
		fontSize &&  typeof maxFontSize === undefined ? `font-size: ${handleFontSizeProps(fontSize)};` : fontSize && maxFontSize? `font-size: ${handleFontSizeProps(capFontSizeFunc(fontSize, maxFontSize))};`: null,
		textColorFromTheme && handleColorFromTheme ? `color: ${handleColorFromTheme(textColorFromTheme, 1, theme)};` : `color: ${getColor()};`,
		restProps.flex ? `flex: ${restProps.flex};` : null,
		fontWeight ? `font-weight: ${fontWeight};` : null,
		restProps.paddingLeft ? `padding-left: ${handleUnitProps(restProps.paddingLeft)};` : null,
		restProps.paddingTop ? `padding-top: ${handleUnitProps(restProps.paddingTop)};` : null,
		restProps.fontStyle ? `font-style: ${restProps.fontStyle};` : null,
		restProps.paddingRight ? `padding-right: ${handleUnitProps(restProps.paddingRight)};` : null,
		restProps.lineHeight ? `line-height: ${handleUnitProps(restProps.lineHeight)};` : fontSize ? `line-height: ${handleUnitProps(fontSize + 6)};` : null,
		restProps.width ? `width: ${handleUnitProps(restProps.width)};` : null,
		restProps.textAlign ? `text-align: ${restProps.textAlign};` : null,
		restProps.marginTop ? `margin-top: ${handleUnitProps(restProps.marginTop)};` : null,
		restProps.marginBottom ? `margin-bottom: ${handleUnitProps(restProps.marginBottom)};` : null,
		restProps.opacity ? `opacity: ${restProps.opacity};` : null,
		restProps.marginLeft ? `margin-left: ${handleUnitProps(restProps.marginLeft)};` : null,
		restProps.marginRight ? `margin-right: ${handleUnitProps(restProps.marginRight)};` : null,
		restProps.shadowOpacity ? `text-shadow-opacity: ${restProps.shadowOpacity};` : null,
		restProps.shadowColor ? `text-shadow-color: ${restProps.shadowColor};` : null,
		(restProps.shadowOffsetX || restProps.shadowOffsetY) ? `text-shadow-offset: ${handleUnitProps(restProps.shadowOffsetX || 0)} ${handleUnitProps(restProps.shadowOffsetY || 0)};` : null,
		restProps.shadowRadius ? `text-shadow-radius: ${handleUnitProps(restProps.shadowRadius)};` : null,
		restProps.textTransform ? `text-transform: ${restProps.textTransform};` : null,
	]

	const css = cssProperties.filter(Boolean).join("\n")
	// console.log("css", css)


	return css
}


export default getStyleForTextProps