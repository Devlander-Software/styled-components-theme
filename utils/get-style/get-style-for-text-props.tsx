import { FontTypeEnum } from "../../../../../../../herb-design-system-react-native/packages/shared/types/font-type.enum"
import { TextStylePropsWithTheme } from "../../../../../../../herb-design-system-react-native/packages/shared/types/text-style.props"

export const getStyleForTextProps = (props: TextStylePropsWithTheme): string => {
	const { theme, fontSize, textColorFromTheme, fontType, fontTypeWeight, fontWeight, ...restProps } = props

	const handleUnitProps = theme?.handleUnitProps ?? null
	const handleColorFromTheme = theme?.handleColorFromTheme ?? null
	const handleFontSizeProps = theme?.handleFontSizeProps ?? null
	const handleFontFromTheme = theme?.handleFontFromTheme ?? null

	const getColor = () => {
		if (restProps.onDark) {
			return theme.colors.whiteAlpha100
		}
		if (restProps.highlight) {
			return theme.colors.accent
		}
		if (textColorFromTheme && theme.colors[textColorFromTheme]) {
			return theme.colors[textColorFromTheme]
		}
		if (restProps.destructive) {
			return theme.colors.errorColor
		}
		return theme.colors.blackAlpha100
	}

	const cssProperties = [
		fontType && fontTypeWeight && handleFontFromTheme ? `font-family: ${handleFontFromTheme(fontType, fontTypeWeight, theme)};` : `font-family: ${handleFontFromTheme(FontTypeEnum.Font1, "bold", theme)};`,
		restProps.textDecorationLine && restProps.textDecorationLine !== "none" ? `text-decoration-line: ${restProps.textDecorationLine};` : null,
		fontSize && handleFontSizeProps ? `font-size: ${handleFontSizeProps(fontSize)};` : null,
		textColorFromTheme && handleColorFromTheme ? `color: ${handleColorFromTheme(textColorFromTheme, 1, theme)};` : `color: ${getColor()};`,
		restProps.flex ? `flex: ${restProps.flex};` : null,
		fontWeight ? `font-weight: ${fontWeight};` : null,
		restProps.paddingLeft ? `padding-left: ${handleUnitProps(restProps.paddingLeft)};` : null,
		restProps.paddingTop ? `padding-top: ${handleUnitProps(restProps.paddingTop)};` : null,
		restProps.fontStyle ? `font-style: ${restProps.fontStyle};` : null,
		restProps.paddingRight ? `padding-right: ${handleUnitProps(restProps.paddingRight)};` : null,
		restProps.lineHeight ? `line-height: ${handleUnitProps(restProps.lineHeight)};` : fontSize ? `line-height: ${handleUnitProps(fontSize + 6)};` : null,
		restProps.width ? `width: ${handleUnitProps(restProps.width)};` : null,
		restProps.textAlign ? `text-align: ${restProps.textAlign};` : null,
		restProps.marginTop ? `margin-top: ${handleUnitProps(restProps.marginTop)};` : null,
		restProps.marginBottom ? `margin-bottom: ${handleUnitProps(restProps.marginBottom)};` : null,
		restProps.opacity ? `opacity: ${restProps.opacity};` : null,
		restProps.marginLeft ? `margin-left: ${handleUnitProps(restProps.marginLeft)};` : null,
		restProps.marginRight ? `margin-right: ${handleUnitProps(restProps.marginRight)};` : null,
		restProps.shadowOpacity ? `text-shadow-opacity: ${restProps.shadowOpacity};` : null,
		restProps.shadowColor ? `text-shadow-color: ${restProps.shadowColor};` : null,
		(restProps.shadowOffsetX || restProps.shadowOffsetY) ? `text-shadow-offset: ${handleUnitProps(restProps.shadowOffsetX || 0)} ${handleUnitProps(restProps.shadowOffsetY || 0)};` : null,
		restProps.shadowRadius ? `text-shadow-radius: ${handleUnitProps(restProps.shadowRadius)};` : null,
		restProps.textTransform ? `text-transform: ${restProps.textTransform};` : null,
	]

	const css = cssProperties.filter(Boolean).join("\n")
	console.log("css", css)


	return css
}
