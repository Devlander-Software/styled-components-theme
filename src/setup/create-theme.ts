import { defaultFonts } from '../defaults/fonts.defaults';
import { lightColors } from '../defaults/light-colors.defaults';
import { ColorsInterface } from "../types/color.types";
import { DeviceOnTheme } from '../types/device-on-theme.type';
import { FontsInterface } from '../types/fonts.interface';
import { ThemeInterface, ThemeInterfaceForHTML } from '../types/theme.types';
import adjustColor from '../utils/adjust-color';
import capFontSize from '../utils/get-style/cap-font-size';
import getStyleForContainerProps from '../utils/get-style/get-style-for-container-props';
import getStyleForGhostProps from '../utils/get-style/get-style-for-ghost-props';
import getStyleForTextProps from '../utils/get-style/get-style-for-text-props';
import getStyleForTouchableOpacityProps from '../utils/get-style/get-style-for-touchable-opacity-props';
import boxShadowOne from '../utils/get-style/handle-box-shadows';
import handleColorFromTheme from '../utils/get-style/handle-color-from-theme';
import handleFontFromTheme from '../utils/get-style/handle-font-for-theme';
import handleFontSizeProps from '../utils/get-style/handle-font-size-props';
import handleSnackbarColor from '../utils/get-style/handle-snackbar-color';
import handleUnitProps from '../utils/get-style/handle-unit-props';
import getStyleForGhostPropsForWeb from '../utils/get-style/web/get-style-for-ghost-props.web';
import { getStyleForImgPropsForWeb } from '../utils/get-style/web/get-style-for-img-props';
import getStyleForParagraphProps from '../utils/get-style/web/get-style-for-paragraph-props';
import handleColorFromThemeWeb from '../utils/get-style/web/handle-color-for-theme.web';
import handleFontFromThemeWeb from '../utils/get-style/web/handle-font-for-theme.web';



export interface ThemeConfig {
    colorPreferences?: Partial<ColorsInterface>
    fontPreferences?: Partial<FontsInterface>
    deviceOnTheme?: Partial<DeviceOnTheme>,
}

export const createThemeForNative = (config: ThemeConfig): ThemeInterface => {
    const { colorPreferences, fontPreferences, deviceOnTheme } = config
    const theme: ThemeInterface = {
        colors: {
            ...lightColors,
            ...colorPreferences,
        },
        fonts: {
            ...defaultFonts,
            ...fontPreferences,
        },
        handleSnackbarColor: handleSnackbarColor,
        handleFontSizeProps: handleFontSizeProps,
        handleUnitProps: handleUnitProps,
        getStyleForTouchableOpacityProps: getStyleForTouchableOpacityProps,
        getStyleForContainerProps: getStyleForContainerProps,
        handleGhost: getStyleForGhostProps,
        getStyleForTextProps: getStyleForTextProps,
        boxShadowOne: boxShadowOne,
        boxShadowTwo: boxShadowOne,
        boxShadowThree: boxShadowOne,
        handleColorFromTheme: handleColorFromTheme,
        handleFontFromTheme: handleFontFromTheme,
        elevation: {
            level1: "",
            level2: "",
            level3: "",
            level4: "",
            level5: "",
        },
        padding: {
            paddingForScrollView: 0,
            paddingForScrollViewTwo: 0
        },
        deviceOnTheme: {
            
            screenWidth: 0,
            statusBarHeight: 0,
            screenHeight: 0,
            isTablet: false,
            hasNotch: false,
            hasDynamicIsland: false,
            platform: 'ios',
            ...deviceOnTheme,
        },
        capFontSize: capFontSize,
        adjustColor: adjustColor,
    }

    return theme
}



export const createThemeForWeb = (config: ThemeConfig): ThemeInterfaceForHTML => {
    const { colorPreferences, fontPreferences, deviceOnTheme } = config
    const theme: ThemeInterfaceForHTML = {
        colors: {
            ...lightColors,
            ...colorPreferences,
        },
        fonts: {
            ...defaultFonts,
            ...fontPreferences,
        },
        handleSnackbarColor: handleSnackbarColor,
        handleFontSizeProps: handleFontSizeProps,
        handleUnitProps: handleUnitProps,
        getStyleForContainerProps: getStyleForContainerProps,
        handleGhost: getStyleForGhostPropsForWeb,
        getStyleForImageProps: getStyleForImgPropsForWeb,
        getStyleForTextProps: getStyleForParagraphProps,
        boxShadowOne: boxShadowOne,
        boxShadowTwo: boxShadowOne,
        boxShadowThree: boxShadowOne,
        handleColorFromTheme: handleColorFromThemeWeb,
        handleFontFromTheme: handleFontFromThemeWeb,
        elevation: {
            level1: "",
            level2: "",
            level3: "",
            level4: "",
            level5: "",
        },
        padding: {
            paddingForScrollView: 0,
            paddingForScrollViewTwo: 0
        },
        deviceOnTheme: {
            screenWidth: 0,
            statusBarHeight: 0,
            screenHeight: 0,
            isTablet: false,
            hasNotch: false,
            hasDynamicIsland: false,
            platform: 'ios',
            ...deviceOnTheme,
          
        },
        capFontSize: capFontSize,
        adjustColor: adjustColor,

    }

    return theme
}


interface CreateThemeConfig extends ThemeConfig {
    framework: "react-native" | "web"
}



const createTheme = (config: CreateThemeConfig): ThemeInterface | ThemeInterfaceForHTML => {
    if(config.framework === "react-native") {
        return createThemeForNative(config)
    } else {
        return createThemeForWeb(config)
    }
 
}


export default createTheme