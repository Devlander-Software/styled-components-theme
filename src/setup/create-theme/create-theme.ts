/* eslint-disable simple-import-sort/imports */
import { ColorsInterface } from "../../types/color.types";
import { DeviceOnTheme } from "../../types/device-on-theme.type";
import { FontsInterface } from "../../types/fonts.interface";
import { ThemeInterface } from "../../types/theme.types";

import { createThemeForNative } from "./create-theme.native";
import { createThemeForWeb } from "./create-theme.web";

export interface ThemeConfig {
  colorPreferences?: Partial<ColorsInterface>;
  fontPreferences?: Partial<FontsInterface>;
  deviceOnTheme?: Partial<DeviceOnTheme>;
}

export interface CreateThemeFunc {
  (config: ThemeConfig): ThemeInterface;
}

export const createTheme: CreateThemeFunc =
  createThemeForNative || createThemeForWeb;

export default createTheme;
