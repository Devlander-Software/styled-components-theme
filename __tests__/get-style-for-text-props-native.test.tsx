import { createThemeForNative } from '../src/bundle/native/setup/create-theme/create-theme.native';
import { getStyleForTextPropsForNative } from '../src/bundle/native/utils/theme-functions/get-style-for-text-props.native';
import { NativeTheme } from '../src/bundle/shared/types/base-theme-types';
import { ResolvedThemedTextStylingProps } from '../src/bundle/shared/types/text-style.types';

jest.mock('react-native', () => {
    return {
        Platform: {
            OS: 'ios', // or 'android', 'web', etc., as needed for your tests
            select: jest.fn((spec) => (spec.ios ? spec.ios : spec.android)),
        },
        // Mock other react-native exports as needed
    };
});

describe('getStyleForTextPropsForNative Function', () => {
    const mockTheme = createThemeForNative({});
    const normalizeCss = (str: string) => str.replace(/\s/g, '');

    it('should generate correct CSS for valid input with theme colors', () => {

        const props: ResolvedThemedTextStylingProps<NativeTheme, number> = {
            theme: mockTheme,
            fontSize: 24,
            textColorFromTheme: 'blackAlpha100',
            fontType: 'Font1',
            fontTypeWeight: 'bold',
            // Add other necessary props here
        };

        const expectedCSS = `
          font-family: Montserrat-Bold;
          color: rgba(0, 0, 0, 1);
          line-height: 30px;
          font-size: 24px;
        `.trim();

        const generatedCSS = getStyleForTextPropsForNative(props as any);

        expect(normalizeCss(generatedCSS)).toEqual(normalizeCss(expectedCSS));
        expect(generatedCSS).toMatchSnapshot();

    });

    it('should prefer the color prop over the textColorFromTheme prop to allow a user to override the color', () => {

        const props: ResolvedThemedTextStylingProps<NativeTheme, number> = {
            theme: mockTheme,
            fontSize: 24,
            textColorFromTheme: 'blackAlpha100',
            fontType: 'Font1',
            color: 'red',
            fontTypeWeight: 'bold',
            // Add other necessary props here
        };

        const expectedCSS = `
            font-family: Montserrat-Bold;
            color: red;
            line-height: 30px;
            font-size: 24px;
        `.trim();

        const generatedCSS = getStyleForTextPropsForNative(props as any);


        expect(normalizeCss(generatedCSS)).toEqual(normalizeCss(expectedCSS));
        expect(generatedCSS).toMatchSnapshot();

    });

    describe('when maxFontSize is specified', () => {

        it('should generate correct CSS with adjusted line-height', () => {

            const props: ResolvedThemedTextStylingProps<NativeTheme, number> = {
                theme: mockTheme,
                fontSize: 16,
                textColorFromTheme: 'blackAlpha100',
                fontType: 'Font1',
                fontTypeWeight: 'bold',
                maxFontSize: 20,
                // Add other necessary props here
            };

            const expectedCSS = `
              font-family: Montserrat-Bold;
              color: rgba(0, 0, 0, 1);
              line-height: 22px;
              font-size: 16px;
            `;

            const generatedCSS = getStyleForTextPropsForNative(props as any);

            expect(normalizeCss(generatedCSS)).toEqual(normalizeCss(expectedCSS));
            expect(generatedCSS).toMatchSnapshot();

        });
    });

});
