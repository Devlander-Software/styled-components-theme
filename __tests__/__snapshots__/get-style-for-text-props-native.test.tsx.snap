// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStyleForTextPropsForNative generates correct CSS for valid input 1`] = `
"font-family: Montserrat-Bold;
color: rgba(0, 0, 0, 1);
line-height: 30px;
font-size: 24px;"
`;

exports[`getStyleForTextPropsForNative getStyleForTextPropsForNative with maxFontSize generates correct CSS for valid input 1`] = `
"font-family: Montserrat-Bold;
color: rgba(0, 0, 0, 1);
line-height: 22px;
font-size: 16px;"
`;
